File: src/main/java/com/mikea/gae/rx/RxServlet.java
package com.mikea.gae.rx;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.mikea.util.Loggers;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.logging.Logger;

@Singleton
class RxServlet extends HttpServlet {
    private final RxImpl rx;

    @Inject
    public RxServlet(RxImpl rx) {
        this.rx = rx;
    }

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        handleRequest(req, resp);
    }

    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        handleRequest(req, resp);
    }

    private void handleRequest(HttpServletRequest request, HttpServletResponse response) {
        rx.handleRequest(request, response);
    }
}

File: src/main/scala/com/mikea/gae/rx/RxServlet.scala
package com.mikea.gae.rx

import com.google.inject.Inject
import com.google.inject.Singleton
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Singleton class RxServlet @Inject() (rx : RxImpl) extends HttpServlet {
  protected override def doGet(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
    handleRequest(req, resp)
  }

  protected override def doPost(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
    handleRequest(req, resp)
  }

  private def handleRequest(request: HttpServletRequest, response: HttpServletResponse): Unit = {
    rx.handleRequest(request, response)
  }
}